Unittest examples for sampleSel

mdkeehan opened the pull request:
1.  Unit tests to show the "golden path" or a minimum working example of how to call sampleSel with 1 and 2 traits. These test catch the exceptions which are being thrown and hence will need to be modified.
2. A function to clear out the globals so the tests have a known static initial state.
3. Another place that possibly needs yesterdays fix.

mdkeehan committed:
clearGlobals for a known state in testing

mdkeehan committed:
another place that needs Hao's fix

mdkeehan committed:
more tests to show golden path failures

mdkeehan committed:
if founder size too small then invlaid array indexes are generated

mdkeehan committed:
mark location of furthur multiple trait locations for review

mdkeehan commented on src/output/phenotype.jl:
```diff
@@ -19,7 +19,8 @@ function getOurPhenVals(my::Cohort, varRes=0)
     for (i,animal) = enumerate(my.animalCohort)
         if length(animal.phenVal) == 0
-            animal.phenVal =  animal.genVal + LRes * randn(nTraits)
+	    #for t in 1:nTraits?
```
Some multi-trait support needed here

reworkhow replied:
Hi Mike,

I added an example here 
https://github.com/reworkhow/XSim.jl/wiki/Multi-trait-Simulation

Hao

mdkeehan commented on src/output/phenotype.jl:
```diff
@@ -45,7 +46,10 @@ function getOurGenVals(my::Cohort,nTraits=0)
             getMyHaps(animal)
             myGenotypes = getMyGenotype(animal)
             #animal.genVal = dot(myGenotypes[common.G.qtl_index],common.G.qtl_effects)
-            animal.genVal = (myGenotypes[common.G.qtl_index]'common.G.qtl_effects)'
```
Again some multi-trait support needed here.

mdkeehan commented:
Hi Hao,

Thanks for the example documentation it really helps.
I have updated my unit tests to follow the example and submitted a pull
request.

Mike

On Thu, Apr 23, 2020 at 6:41 PM Hao Cheng <notifications@github.com> wrote:

> *@reworkhow* commented on this pull request.
> ------------------------------
>
> In src/output/phenotype.jl
> <https://github.com/reworkhow/XSim.jl/pull/12#discussion_r413549098>:
>
> > @@ -19,7 +19,8 @@ function getOurPhenVals(my::Cohort, varRes=0)
>      genVals  = getOurGenVals(my,nTraits)
>      for (i,animal) = enumerate(my.animalCohort)
>          if length(animal.phenVal) == 0
> -            animal.phenVal =  animal.genVal + LRes * randn(nTraits)
> +	    #for t in 1:nTraits?
>
> Hi Mike,
>
> I added an example here
> https://github.com/reworkhow/XSim.jl/wiki/Multi-trait-Simulation
>
> Hao
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/reworkhow/XSim.jl/pull/12#discussion_r413549098>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AFAOG5IYSYOVPKBBGKDS5E3RN7PJRANCNFSM4MKNJ2CA>
> .
>
